### PR TITLE
test: stop the headless dispatcher flake with per-assembly isolation

### PR DESCRIPTION
**Pull request by yuto-trd**
## Description

- `Beutl.HeadlessUITests` inherited Avalonia's default `PerTest` isolation, which tears the `Application` down after every test and leaves the static `Dispatcher.UIThread` null with no `IDispatcherImpl` registered. This suite keeps editor state alive across tests on purpose (see `TestReset`), so a service still running from the previous test could re-create that singleton as the run-loop-less `NullDispatcherImpl`. The next test whose body did not complete synchronously then died inside `HeadlessUnitTestSession` at `Dispatcher.PushFrame` with `PlatformNotSupportedException`, and the test after it reset the field again — hence exactly one rotating victim per run, with an identical stack every time. Three of the last four `.NET` CI failures on `main` were this.
- `PerAssembly` isolation never nulls the dispatcher, so the window disappears. Sharing one `Application` for the whole run then exposes tests that flip `RequestedThemeVariant` and never restore it, so those now restore it — otherwise the first theme test decides the theme for every test after it (`DockTabAddButtonTests` failed deterministically on that).
- Two of the restorations were cleanup blocks in `DarkBorderThemeExtensionTests` that pinned the variant to `Dark` in the name of cleanup; they now restore the value captured on entry. `ThemeServiceTests`' existing `ThemeScope` already restored merged dictionaries and the accent, and now restores the variant too.

## Affected areas
- [ ] `Beutl.Engine` (rendering / scene / track)
- [ ] `Beutl.ProjectSystem` (project / document persistence)
- [ ] UI (`Beutl.Editor`, `Beutl.Editor.Components`, `Beutl.Controls`)
- [ ] `Beutl.Extensibility` (plugin abstractions)
- [ ] `Beutl.NodeGraph` (node editor)
- [ ] `Beutl.FFmpegIpc` / `Beutl.FFmpegWorker` (media IPC boundary)
- [ ] `Beutl.Api` (server API client)
- [x] Build / CI / docs only

Test-infrastructure only — every changed file is under `tests/Beutl.HeadlessUITests/`. No production code is touched.

## Breaking changes

None.

## Test plan

- **Root cause confirmed, not inferred.** A temporary NUnit `ITestAction` reflected Avalonia's private `Dispatcher.s_uiThread` before each test (reflection rather than the `Dispatcher.UIThread` property, which would have created the singleton and caused the very bug). Every captured failure showed `SupportsRunLoops=False` on the failing test and `True` on the test before it. The diagnostic is not part of this PR.
- **Flake reproduction, macOS host:** looping `--filter FullyQualifiedName~EditorProjectSessionGatewayTests` failed **2 of 4** runs before the change and **0 of 40** after.
- **Full assembly:** 199/199 green across repeated runs; `dotnet build` 0 warnings / 0 errors; `dotnet format --verify-no-changes` clean.
- **Regression guard:** `HeadlessSessionIsolationTests` asserts the assembly declares `PerAssembly` isolation (deterministic — fails immediately if the attribute is dropped) and that `Dispatcher.UIThread.SupportsRunLoops` holds, which is the precondition `PushFrame` checks.
- Not reproducible in the arm64 SwiftShader container (60 iterations, 0 hits); use a macOS host to reproduce.

## Fixed issues / References

- CI runs with the signature fixed here: 30438903976, 30600715402.
- https://github.com/b-editor/beutl/actions/runs/31229274598 was killed by a separate native test-host crash (`Reason: Test host process crashed`, no managed stack), which this PR reduces but does not eliminate. That one is a distinct defect and is being tracked separately with the maintainer, not fixed here.

## Summary by CodeRabbit

* **Tests**
  * Improved headless UI test isolation and dispatcher run-loop coverage.
  * Added safer theme handling so tests restore the previously active theme after execution.
  * Updated dark-theme UI tests to apply themes only within their individual test scopes.
  * Preserved application and editor state consistently across related test runs.

### CODE REVIEW

File: tests/Beutl.HeadlessUITests/ElementViewLayoutTests.cs [hunk 1]
```diff
@@ -74,6 +73,7 @@ private static void AddRectangle(EditViewModel editor)
     [AvaloniaTest]
     public async Task Label_is_inset_clear_of_the_rounded_corner()
     {
+        using ThemeVariantScope themeScope = ThemeVariantScope.Use(ThemeVariant.Dark);
```
[the remark this answers is not in the document]
Comment (yuto-trd) [reply tests/Beutl.HeadlessUITests/ElementViewLayoutTests.cs:76]: Not applicable here — `TestReset.ResetShellAsync()` is already the first statement of `InflateFirstElementView` (`tests/Beutl.HeadlessUITests/ElementViewLayoutTests.cs:60`), which each of the five tests awaits from inside its `[AvaloniaTest]` body. The reset therefore still runs in the body, on the Avalonia UI thread, before any `TestShell` access.

The guideline this is measured against (`tests/CLAUDE.md`) requires the reset to happen in the test body rather than in `[SetUp]`/`[TearDown]` — those run off the UI thread, where mutating Avalonia state silently fails — not to be the literal first statement.

The only statement preceding it is `ThemeVariantScope.Use(ThemeVariant.Dark)`, which assigns `Application.Current.RequestedThemeVariant` and nothing else (`tests/Beutl.HeadlessUITests/ThemeVariantScope.cs:16-17`); it touches no shell singleton, so there is no ordering dependency on the reset. `InflateFirstElementView` also has no callers outside these five tests, so hoisting the reset into each body would duplicate it with no behavioural change. Leaving as is.
